tools/rtpsrelay/lib/Name.cpp

```diff
@@ -0,0 +1,179 @@
+#include "Name.h"
+
+#include <ostream>
+
+static void output_character(std::ostream& out, char c) {
+  switch (c) {
+  case '[':
+  case ']':
+  case '!':
+  case '\\':
+  case '?':
+  case '*':
+    out << '\\';
+  }
+  out << c;
+}
+
+std::ostream& operator<<(std::ostream& out, const Atom& atom)
+{
+  switch (atom.kind()) {
+  case Atom::CHARACTER:
+    output_character(out, atom.character());
+    break;
+  case Atom::CHARACTER_CLASS:
+    out << '[';
+    for (char c : atom.characters()) {
+      output_character(out, c);
+    }
+    out << ']';
+    break;
+  case Atom::NEGATED_CHARACTER_CLASS:
+    out << "![";
+    for (char c : atom.characters()) {
+      output_character(out, c);
+    }
+    out << ']';
+    break;
+  case Atom::WILDCARD:
+    out << '?';
+    break;
+  case Atom::GLOB:
+    out << '*';
+    break;
+  }
+
+  return out;
+}
+
+void Name::parse(Name& name, const std::string& buffer, size_t& idx)
+{
+  if (!name.is_valid_ || idx == buffer.length()) {
```

Comment:
```suggestion
  if (!name.is_valid_ || idx == buffer.size()) {
```
`size` is preferred since all std library containers have `size`